File: CBASIC/UtilFiles/StudentGrading.js
// Student Grading System

function gradingSystem() {
    const maxStudents = 50;
    const studentNames = [];
    const grades = [];
    
    let numStudents = parseInt(prompt("Enter the number of students (max 50):"), 10);
    if (numStudents > maxStudents) {
        console.log(`Number of students cannot exceed ${maxStudents}.`);
        return;
    }

    for (let i = 0; i < numStudents; i++) {
        let studentName = prompt(`Enter the name of student ${i + 1}:`);
        let grade = parseFloat(prompt(`Enter the grade of ${studentName}:`));
        studentNames.push(studentName);
        grades.push(grade);
    }

    let total = grades.reduce((acc, grade) => acc + grade, 0);
    let average = total / numStudents;

    console.log(`Average Grade: ${average.toFixed(2)}`);
    console.log("Student Grades:");

    for (let i = 0; i < numStudents; i++) {
        let letterGrade = "";
        if (grades[i] >= 90) {
            letterGrade = "A";
        } else if (grades[i] >= 80) {
            letterGrade = "B";
        } else if (grades[i] >= 70) {
            letterGrade = "C";
        } else if (grades[i] >= 60) {
            letterGrade = "D";
        } else {
            letterGrade = "F";
        }

        console.log(`${studentNames[i]}: ${grades[i]} - ${letterGrade}`);
    }
}

// Run the grading system
gradingSystem();

File: CBASIC/UtilFiles/StudentGrading.py
# Student Grading System

def grading_system():
    max_students = 50
    student_names = []
    grades = []

    num_students = int(input("Enter the number of students (max 50): "))
    if num_students > max_students:
        print(f"Number of students cannot exceed {max_students}.")
        return

    for i in range(num_students):
        student_name = input(f"Enter the name of student {i + 1}: ")
        grade = float(input(f"Enter the grade of {student_name}: "))
        student_names.append(student_name)
        grades.append(grade)

    total = sum(grades)
    average = total / num_students

    print(f"Average Grade: {average:.2f}")
    print("Student Grades:")
    
    for i in range(num_students):
        letter_grade = ""
        if grades[i] >= 90:
            letter_grade = "A"
        elif grades[i] >= 80:
            letter_grade = "B"
        elif grades[i] >= 70:
            letter_grade = "C"
        elif grades[i] >= 60:
            letter_grade = "D"
        else:
            letter_grade = "F"

        print(f"{student_names[i]}: {grades[i]} - {letter_grade}")

# Run the grading system
grading_system()
